File: internal/models/usuario.go
package models

import (
	"gorm.io/gorm"
)

type Usuario struct {
	gorm.Model

	ID         uint   `gorm:"primaryKey"`
	Email      string `gorm:"unique;not null" json:"email"`
	Contrasena string `gorm:"not null" json:"contrasena"`
	Nombre     string `gorm:"not null" json:"nombre"`
	Rol        string `gorm:"default:'usuario'" json:"rol"`
}

File: src/main/java/com/cine/cinema/models/entities/usuario/Usuario.java
package com.cine.cinema.models.entities.usuario;

import lombok.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "users")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long usuarioId;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private Rol rol = Rol.CLIENTE;

    // ID de Keycloak (puede ser UUID, pero lo pides como Date)
    @Column(name = "keycloak_id")
    private java.util.Date keycloakId;
}
